src/dialog-user/services/dialogData.ts

```diff
@@ -33,9 +33,11 @@ export default class DialogDataService {
     options,
     values,
   }) {
+    // single value selects do contain [null, '<None>']
+    const allIntegers = values.every((item) => (item[0] === null ? true : !isNaN(item[1]) && Number.isInteger(Number(item[1]))));
```

Comment:
`allIntegers` is true when all descriptions except for the null item look like a number

previous behaviour would only treat actual numbers as numbers, is the change intentional?

Sounds like this would leaad to us sorting descriptions numerically whenever data_type is set to integer,
where previously we could only sort descriptions numerically when they were coming from dynamic fields, if ever.

Reply:
I've been back & forth here and in truth, I don't know how the sorting should behave. But since this PR is about fixing dialog editor and friends so that it honors sorting parameters and not about changing sorting behavior, I'm reverting my change to the original.